internal/deploy/values/builder.go

```diff
@@ -83,16 +85,21 @@ func (b *builder) mergeSources() (map[string]interface{}, error) {
 		}
 
 		if err := mergo.Map(&result, vals, mergo.WithOverride); err != nil {
-			return nil, err
+			return nil, errors.Wrap(err, "while merging values from files")
 		}
 	}
 
 	for _, override := range b.values {
 		if err := mergo.Map(&result, override, mergo.WithOverride); err != nil {
-			return nil, err
+			return nil, errors.Wrap(err, "while merging overrides values")
 		}
 	}
 
+	err := mergo.Map(&result, b.dockerRegistryOverrides)
+	if err != nil {
+		return nil, errors.Wrap(err, "while merging docker registry overrides")
+	}
+
```

Comment:
Let's move it up (before values and files with `mergo.Overrides`) to keep the logical flow:
1. First we apply defaults
2. Then we add files
3. Then we add values